app/controllers/foods_controller.rb

```diff
@@ -1,3 +1,33 @@
 class FoodsController < ApplicationController
-  def index; end
+  def new
+    @user = current_user
+    @food = Food.new
+  end
+
+  def create
+    @food = current_user.foods.new(food_params)
+    if @food.save
+      redirect_to foods_path
+    else
+      render :new
```

Comment:
It would be good to let the user know what errors it is having while filling the form, try using:
```rb
flash.now[:alert] = @food.errors.full_messages.to_sentence
```
before rendering once again the new form.